src/subsystems/haskell/builders/default/default.nix

```diff
@@ -35,10 +35,18 @@ in {
     # Example:
     #   produceDerivation name (mkDerivation {...})
     produceDerivation,
+    compiler ? null,
```

Comment:
Actually there is not need to have this as an argument. There is no UI to set these arguments